### PR TITLE
New formula: vulkan-headers 1.1.99

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/vulkan-headers.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class VulkanHeaders < Formula
+  desc "Vulkan Header files and API registry"
+  homepage "https://github.com/KhronosGroup/Vulkan-Headers"
+  url "https://github.com/KhronosGroup/Vulkan-Headers/archive/v1.1.99.tar.gz"
+  sha256 "37506deaf56b254193e4fa83195add2617e881efcac3d2ff5f0cf49353eb9aa8"
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/vulkan-headers.rb]: Blank line above `depends_on`

File: Formula/vulkan-headers.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    assert_predicate "#{include}/vulkan/vulkan.h", :exist?
```
Comment (fxcoudert) [inline Formula/vulkan-headers.rb]: We need to compile a test program against the headers, and run it.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/vulkan-headers.rb (hunk 2 above)
Comment (yurikoles) [reply Formula/vulkan-headers.rb]: But we are unable to link it, without actual libs.

Comment (yurikoles) [reply Formula/vulkan-headers.rb]: How about to try tell `cc` to just produce `o` file without linking the binary?

**Comment by yurikoles** [comment]
@MikeMcQuaid what is a proper way to check headers-only formula?

**Comment by fxcoudert** [comment]
Is there absolutely no header that can be tested without linking to a specific library? Like a version header or something like that?

**Comment by MikeMcQuaid** [comment]
We historically have rejected header-only formulae and said they should be vendored in projects instead. If we do include this we'll need something that `#include`s the header and uses the code.

**Comment by yurikoles** [comment]
@MikeMcQuaid @fxcoudert 
This formula is first step to bring up MoltenVK and other Vulkan-related formulas. 
In #30465 MoltenVK was rejected because we need to fetch deps manually, but there is also option to use existing ones, like this formula provides.

**Comment by fxcoudert** [comment]
Seems like a good reason enough.

**Review by fxcoudert — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Make sense :+1:

**Comment by fxcoudert** [comment]
Merged, thanks @

**Pull request closed without merging**